modules/access_level/main.tf

```diff
@@ -32,14 +32,18 @@ resource "google_access_context_manager_access_level" "access_level" {
       members                = var.members
       negate                 = var.negate
 
-      device_policy {
-        require_screen_lock              = var.require_screen_lock
-        allowed_encryption_statuses      = var.allowed_encryption_statuses
-        allowed_device_management_levels = var.allowed_device_management_levels
+      dynamic "device_policy" {
+        for_each = var.device_policy_enabled ? [{}] : []
```

Comment:
Instead of introducing a new variable, I think we can compute this value.

Ex:

```
require_screen_lock || length(allowed_encryption_statuses) >= 0 || length(allowed_device_management_levels) >= 0 || minimum_version != "" || os_type != "OS_UNSPECIFIED
```

Reply:
I made the change you had suggested. I had to make a few minor changes. I tested it locally. It seems to be working however some integration tests seem to be failing. I don't have any visibility into why the tests are failing. Let me know if I need to make any more changes. Thanks

Reply:
FYI you should be able to run the integration tests locally, but I'll take a look.

Reply:
@morgante  I went through the contribution guide to run the integration tests locally. It required using a service account with folder level permissions. It also has access. context manager policy admin permissions.

When I run 
make docker_test_perpare

Error: error creating project ci-project-policy-test-0-dbc5 (ci-project-policy-test-0): googleapi: Error 409: Requested entity already exists, alreadyExists. If you received a 403 error, make sure you have the `roles/resourcemanager.projectCreator` permission

  on .terraform/modules/project-vpc-service-controls-policy-0/modules/core_project_factory/main.tf line 126, in resource "google_project" "main":
 126: resource "google_project" "main" {

I did give the service account "Project Creator"  Project Creator Permissions before running the make command.
I will keep on debugging this error.

Thanks for your help